server/handlers/validation.go

```diff
@@ -80,9 +80,21 @@ func validateUpdate(cs signed.CryptoService, gun string, updates []storage.MetaU
 			return nil, ErrBadTargets{msg: err.Error()}
 		}
 		repo.SetTargets(targetsRole, t)
+	} else {
+		targetsJSON, err := store.GetCurrent(gun, data.CanonicalTargetsRole)
+		if err != nil {
+			return nil, fmt.Errorf("could not load repo for snapshot generation: %v", err)
```

Comment:
Answering my own question, it seems like it's implied [because when listing the top level metadata files, only mirror is listed as optional](https://github.com/theupdateframework/tuf/blob/develop/docs/tuf-spec.txt#L348).  Although it'd be nice if the spec explicitly said that those files must exist, or that the snapshot metadata must have hashes for all those roles, like [spec definition for the root.json metadata says it MUST have keys for all 4 roles](https://github.com/theupdateframework/tuf/blob/develop/docs/tuf-spec.txt#L519).  So it seems like maybe they're leaving the implementation open, but we can just interpret it as it always has to exist, even if it's empty?